File: concepts/src/test/java/SinglyListNodeTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SinglyListNodeTest {
    @Test
    public void sum() {
        assertEquals(
            0,
            SinglyListNode.ZERO
                .stream()
                .mapToInt(n -> n.value)
                .sum()
        );
        assertEquals(
            16,
            SinglyListNode
                .of(1, 3, 5, 7)
                .stream()
                .mapToInt(n -> n.value)
                .sum()
        );
    }

    @Test
    public void traverse() {
        ListNode<SinglyListNode> node = SinglyListNode.of(2, 4, 6, 8);
        assertEquals(2, node.value);
        node = node.next.next;
        assertEquals(6, node.value);
        node = node.next;
        assertEquals(8, node.value);
    }
}

File: javascript/__tests__/concepts/singly-list-node.test.js
import {strictEqual} from 'assert';
import SinglyListNode, {SINGLY_LIST_NODE_ZERO} from '../../src/concepts/singly-list-node';

describe(
    'SinglyListNode',
    () => {
      test(
          'sum',
          () => {
            let total = 0;
            for (const n of SINGLY_LIST_NODE_ZERO) {
              total += n.val;
            }
            strictEqual(0, total);

            total = 0;
            for (const n of SinglyListNode.of(1, 3, 5, 7)) {
              total += n.val;
            }
            strictEqual(16, total);
          },
      );
      test(
          'traverse',
          () => {
            let node = SinglyListNode.of(2, 4, 6, 8);
            strictEqual(2, node.val);

            node = node.next.next;
            strictEqual(6, node.val);

            node = node.next;
            strictEqual(8, node.val);
          },
      );
    },
);
